### PR TITLE
Add branch name checks using husky

### PR DESCRIPTION
**Pull request by mohitb35**
Introduces pre-commit hooks using husky to check that the branch name conforms to the following rules:

1. Permanent branches - `main` `develop`
2. Temporary branches start with
     a. `feature/` - most changes will use this.
     b. `hotfix/` - changes that need to be merged directly to main will use this.
3. Permitted characters (after /) - lowercase alphabets, numbers, hyphen/underscore as separator

UPDATE: updated ESL rules to ignore ESL error causing build failure on vercel (due to dependency issue).
```
Unknown property 'jsx' found react/no-unknown-property 
Unknown property 'global' found react/no-unknown-property 
```
Ref: https://github.com/vercel/next.js/discussions/40269

### CODE REVIEW

**Review by norbertschuler — APPROVED** [review]
LGFM, please also add to planet-webapp. Old PR's with other names probably should not be merging develop into it (once this is merged into develop) or have to be renamed then to be continued.

**Comment by mohitb35** [comment]
duplicate to #369 - should be closed.

File: package.json [hunk 1]
```diff
@@ -9,7 +9,12 @@
     "lint": "eslint pages src",
     "lint:errors": "eslint pages src --quiet",
     "cypress:open": "cypress open",
-    "cypress:run": "cypress run"
+    "cypress:run": "cypress run",
+    "prepare": "husky install"
```
Comment (norbertschuler) [inline package.json:13]: I know we do not deploy planet-donations anywhere else but with Vercel, but do you want to make the same change as for planet-webapp to support other deployments in the future disabling this check for production releases? https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/develop/package.json#L24

**Comment by mohitb35** [comment]
Yes, I'll do that here as well

Regards
Mohit
________________________________

File: package.json (hunk 1 above)
Comment (mohitb35) [reply package.json]: Done

**Review by norbertschuler — APPROVED** [review]

**Pull request merged**